apps/admin-gui/src/app/shared/components/assign-groups-sponsored-members/assign-groups-sponsored-members-component.ts

```diff
@@ -0,0 +1,181 @@
+  assignableGroups: Group[] = [];
+  allVoGroups: Group[] = [];
+  manualMemberAddingBlocked = false;
+  groupAssignment = 'none';
+
+  selection = new SelectionModel<Group>(true, []);
+  name = '';
+  description = '';
+  asSubgroup = false;
+  parentGroup: Group = null;
+  filterValue = '';
+  tableId = TABLE_VO_GROUPS;
+  groupToCreate: Group = {
+    id: 0,
+    beanName: 'Group',
+  };
+  groupIds: number[] = [];
+
+  private groupAttrNames = [Urns.GROUP_SYNC_ENABLED, Urns.GROUP_BLOCK_MANUAL_MEMBER_ADDING];
+
+  constructor(
+    private groupsService: GroupsManagerService,
+    private guiAuthResolver: GuiAuthResolver,
+    private attributesService: AttributesManagerService
+  ) {}
+
+  ngOnInit(): void {
+    this.submit.subscribe(() => this.handleGroups());
+    this.selection.changed.subscribe(() => this.isSubmitAllowed());
+    this.createGroupAuth = this.guiAuthResolver.isAuthorized('createGroup_Vo_Group_policy', [
+      { id: this.voId, beanName: 'Vo' },
+    ]);
+    this.attributesService.getVoAttributes(this.voId).subscribe((attributes) => {
+      this.manualMemberAddingBlocked = hasBooleanAttributeEnabled(
+        attributes,
+        Urns.VO_BLOCK_MANUAL_MEMBER_ADDING
+      );
+      this.groupsService
+        .getAllRichGroupsWithAttributesByNames(this.voId, this.groupAttrNames)
+        .subscribe((groups) => {
+          this.allVoGroups = groups.filter((grp) => grp.name !== 'members');
+          this.assignableGroups = this.filterAssignableGroups(groups);
+          this.isSubmitAllowed();
+        });
+    });
+  }
+
+  groupAssigmentChanged(): void {
+    this.selection.clear();
+
+    this.groupToCreate.name = '';
+    this.groupToCreate.description = '';
+    this.asSubgroup = false;
+    this.groupToCreate.parentGroupId = null;
+    this.isSubmitAllowed();
+  }
+
+  applyFilter(filterValue: string): void {
+    this.filterValue = filterValue;
+  }
+
+  onNameChange(name: string): void {
+    this.groupToCreate.name = name;
+    this.isSubmitAllowed();
+  }
+
+  onParentChange(parentGroup: Group): void {
+    if (parentGroup === null) {
+      this.groupToCreate.parentGroupId = null;
+    } else {
+      this.groupToCreate.parentGroupId = parentGroup.id;
+    }
+    this.isSubmitAllowed();
+  }
+
+  onAsSubgroupChange(asSubgroup: boolean): void {
+    this.asSubgroup = asSubgroup;
+    this.isSubmitAllowed();
+  }
+
+  onDescriptionChange(description: string): void {
+    this.groupToCreate.description = description;
+    this.isSubmitAllowed();
+  }
+
+  isSubmitAllowed(): void {
+    this.submitAllowed.emit(
+      !(
+        this.groupAssignment === null ||
+        (this.groupAssignment === 'existing' && this.selection.selected.length === 0) ||
+        (this.groupAssignment === 'new' &&
+          (this.groupToCreate.name.length === 0 || this.groupToCreate.description.length === 0)) ||
+        (this.asSubgroup && this.groupToCreate.parentGroupId === null)
+      )
+    );
+  }
+
+  handleGroups(): void {
```

Comment:
I am not sure where exactly, but probably somewhere in this method you should clear the groups (group Ids) when you submit the last step (or on an error) because when for example user type an already existing login, the last step will fail, but then if I try it again to add to another group, for example, the previous group is also presented (tested on the use case when I tried to create new group, failed on login, try again with correct login BUT without group -> failed that group already exists)

Reply:
When an error occurred and the stepper reloaded, a new instance of the groups component got initialized, hence duplicating the subscription to the submit button (and the groups). Added onDelete method that should deal with it